src/utils.cc

```diff
@@ -219,7 +219,7 @@ int utils::SystemCallNotExitOnFailed(int return_value) {
 int utils::SystemCallExitOnFailed(int return_value) {
   if (return_value == ERROR_CODE_SYSTEM) {
     utils::PrintSystemError(std::cerr);
-    exit(ERROR_CODE_DEFAULT);
+    _exit(ERROR_CODE_DEFAULT);
```

Comment:
`SystemCallExitOnFailed ` should use `exit` instead of `_exit`, `_exit` only used when `exec` failed. So our `ProcessChild` should look like this:

```c++
utils::SystemCallNoExitOnFailed(
      execvp(command_parse_result.command.c_str(), &argv[0]));
_exit(ERROR_CODE_DEFAULT);
```